src/taipy/core/data/data_node.py

```diff
@@ -441,6 +440,10 @@ def __getitem__(self, item):
     def _read(self):
         raise NotImplementedError
 
+    @abstractmethod
+    def _read_by(self, operators: Union[List, Tuple], join_operator=JoinOperator.AND):
```

Comment:
Why don't you override the `filter` method directly instead of going through a `_read_by` method that is eventually not implemented?
It is possible to override the `filter` method in MongoCollectionDataNode and not in CsvDataNode, for instance.

Reply:
The reason is that I completely forgot 🤣